editor/components/inserter/index.js

```diff
@@ -37,7 +37,7 @@ class Inserter extends wp.element.Component {
 					label={ wp.i18n.__( 'Insert block' ) }
 					onClick={ this.toggle }
 					className="editor-inserter__toggle" />
-				{ opened && <InserterMenu position={ position } onSelect={ this.close } /> }
+				{ opened && <InserterMenu position={ position } onSelect={ this.close } onClickOutside={ this.close } /> }
```

Comment:
The reason I created the second prop is in case there was an instance which we want a different actions for `onSelect` and `onClickOutside`. 

I think we should keep these because I will probably use this to implement to toggle mentioned below.